webhook/workspace/handler/immutable.go

```diff
@@ -164,6 +163,17 @@ func (h *WebhookHandler) checkRestrictedAccessAnnotation(req admission.Request)
 	return annotations[constants.WorkspaceRestrictedAccessAnnotation] == "true", nil
 }
 
+func checkRestrictedWorkspaceMetadata(oldMeta, newMeta *metav1.ObjectMeta) (allowed bool, msg string) {
+	if oldMeta.Labels[constants.WorkspaceCreatorLabel] != newMeta.Labels[constants.WorkspaceCreatorLabel] {
```

Comment:
I wonder why we need it?
We already have a creator label logic(a bit better in terms of UX since it can be restored from old obj if it's missing) on higher level.

But maybe restricted access check should be done after creator annotation, how it does not seem to be a case: https://github.com/amisevsk/devworkspace-operator/blob/allow-annotations-webhooks/webhook/workspace/handler/workspace.go#L57

Reply:
:+1:

Removed the check here and reordered the checks (check creator first)

Reply:
Actually, looking at it a second more: the reason we do the restricted-access check first is because of
```go
	newCreator, found := newWksp.Labels[constants.WorkspaceCreatorLabel]
	if !found {
		newWksp.Labels[constants.WorkspaceCreatorLabel] = oldCreator
		return h.returnPatched(req, newWksp)
	}
```
I've dropped the commit that changed the order.